```
/home/nuclearkatie/.local/include/cyclus/toolkit/res_buf.h: In instantiation of ‘void cyclus::toolkit::ResBuf<T>::Push(std::vector<B>) [with B = boost::shared_ptr<cyclus::Resource>; T = cyclus::Resource]’:
/home/nuclearkatie/cyclus/cycamore/src/sink.cc:32:35:   required from here
/home/nuclearkatie/.local/include/cyclus/toolkit/res_buf.h:331:9: error: no matching function for call to ‘cyclus::Resource::ChangePackageId()’
  331 |         rss[i]->ChangePackageId();
      |         ^~~
In file included from /home/nuclearkatie/.local/include/cyclus/agent.h:15,
                 from /home/nuclearkatie/.local/include/cyclus/context.h:17,
                 from /home/nuclearkatie/.local/include/cyclus/cyclus.h:11,
                 from /home/nuclearkatie/cyclus/cycamore/src/sink.h:9,
                 from /home/nuclearkatie/cyclus/cycamore/src/sink.cc:7:
/home/nuclearkatie/.local/include/cyclus/resource.h:98:16: note: candidate: ‘virtual void cyclus::Resource::ChangePackageId(int)’
   98 |   virtual void ChangePackageId(int new_package_id) = 0;
      |                ^~~~~~~~~~~~~~~
/home/nuclearkatie/.local/include/cyclus/resource.h:98:16: note:   candidate expects 1 argument, 0 provided
In file included from /home/nuclearkatie/.local/include/cyclus/toolkit/matl_buy_policy.h:9,
                 from /home/nuclearkatie/.local/include/cyclus/cyclus.h:51,
                 from /home/nuclearkatie/cyclus/cycamore/src/sink.h:9,
                 from /home/nuclearkatie/cyclus/cycamore/src/sink.cc:7:
/home/nuclearkatie/.local/include/cyclus/toolkit/res_buf.h:335:9: error: no matching function for call to ‘cyclus::Resource::ChangePackageId()’
  335 |         rss[i]->ChangePackageId();
      |         ^~~
In file included from /home/nuclearkatie/.local/include/cyclus/agent.h:15,
                 from /home/nuclearkatie/.local/include/cyclus/context.h:17,
                 from /home/nuclearkatie/.local/include/cyclus/cyclus.h:11,
                 from /home/nuclearkatie/cyclus/cycamore/src/sink.h:9,
                 from /home/nuclearkatie/cyclus/cycamore/src/sink.cc:7:
/home/nuclearkatie/.local/include/cyclus/resource.h:98:16: note: candidate: ‘virtual void cyclus::Resource::ChangePackageId(int)’
   98 |   virtual void ChangePackageId(int new_package_id) = 0;
      |                ^~~~~~~~~~~~~~~
/home/nuclearkatie/.local/include/cyclus/resource.h:98:16: note:   candidate expects 1 argument, 0 provided
In file included from /home/nuclearkatie/.local/include/cyclus/toolkit/matl_buy_policy.h:9,
                 from /home/nuclearkatie/.local/include/cyclus/cyclus.h:51,
                 from /home/nuclearkatie/cyclus/cycamore/src/sink.h:9,
                 from /home/nuclearkatie/cyclus/cycamore/src/sink.cc:7:
/home/nuclearkatie/.local/include/cyclus/toolkit/res_buf.h: In instantiation of ‘void cyclus::toolkit::ResBuf<T>::Push(cyclus::Resource::Ptr) [with T = cyclus::Resource; cyclus::Resource::Ptr = boost::shared_ptr<cyclus::Resource>]’:
/home/nuclearkatie/cyclus/cycamore/src/sink.cc:167:30:   required from here
/home/nuclearkatie/.local/include/cyclus/toolkit/res_buf.h:278:7: error: no matching function for call to ‘cyclus::Resource::ChangePackageId()’
  278 |       m->ChangePackageId();
      |       ^
In file included from /home/nuclearkatie/.local/include/cyclus/agent.h:15,
                 from /home/nuclearkatie/.local/include/cyclus/context.h:17,
                 from /home/nuclearkatie/.local/include/cyclus/cyclus.h:11,
                 from /home/nuclearkatie/cyclus/cycamore/src/sink.h:9,
                 from /home/nuclearkatie/cyclus/cycamore/src/sink.cc:7:
/home/nuclearkatie/.local/include/cyclus/resource.h:98:16: note: candidate: ‘virtual void cyclus::Resource::ChangePackageId(int)’
   98 |   virtual void ChangePackageId(int new_package_id) = 0;
      |                ^~~~~~~~~~~~~~~
/home/nuclearkatie/.local/include/cyclus/resource.h:98:16: note:   candidate expects 1 argument, 0 provided
In file included from /home/nuclearkatie/.local/include/cyclus/toolkit/matl_buy_policy.h:9,
                 from /home/nuclearkatie/.local/include/cyclus/cyclus.h:51,
                 from /home/nuclearkatie/cyclus/cycamore/src/sink.h:9,
                 from /home/nuclearkatie/cyclus/cycamore/src/sink.cc:7:
/home/nuclearkatie/.local/include/cyclus/toolkit/res_buf.h:282:7: error: no matching function for call to ‘cyclus::Resource::ChangePackageId()’
  282 |       m->ChangePackageId();
      |       ^
In file included from /home/nuclearkatie/.local/include/cyclus/agent.h:15,
                 from /home/nuclearkatie/.local/include/cyclus/context.h:17,
                 from /home/nuclearkatie/.local/include/cyclus/cyclus.h:11,
                 from /home/nuclearkatie/cyclus/cycamore/src/sink.h:9,
                 from /home/nuclearkatie/cyclus/cycamore/src/sink.cc:7:
/home/nuclearkatie/.local/include/cyclus/resource.h:98:16: note: candidate: ‘virtual void cyclus::Resource::ChangePackageId(int)’
   98 |   virtual void ChangePackageId(int new_package_id) = 0;
      |                ^~~~~~~~~~~~~~~

--- a/src/material.h
+++ b/src/material.h
@@ -155,11 +155,11 @@ class Material: public Resource {
   /// DEPRECATED - use non-const comp() function.
   Composition::Ptr comp() const;
 
-  int package_id();
+  virtual int package_id();
 
   /// Changes the package id. Checks that the resource fits the package 
   /// type minimum and maximum mass criteria.
-  void ChangePackageId(int new_package_id = default_package_id_);
+  virtual void ChangePackageId(int new_package_id = default_package_id_);
 
  protected:
   Material(Context* ctx, double quantity, Composition::Ptr c, int package_id = default_package_id_);

--- a/src/product.h
+++ b/src/product.h
@@ -71,11 +71,11 @@ class Product : public Resource {
   /// @throws ValueError 'other' resource is of different quality
   void Absorb(Product::Ptr other);
 
-  // Returns the package id.
-  int package_id();
+  /// Returns the package id.
+  virtual int package_id();
 
   /// Changes the product's package id
-  void ChangePackageId(int new_package_id = default_package_id_);
+  virtual void ChangePackageId(int new_package_id = default_package_id_);
 
  private:
   /// @param ctx the simulation context

--- a/src/resource.h
+++ b/src/resource.h
@@ -92,10 +92,10 @@ class Resource {
   virtual void Absorb(Ptr res) { throw Error("cannot absorb resource type " + this->type()); };
 
   /// Returns the package id.
-  virtual int package_id() = 0;
+  virtual int package_id() { return default_package_id_; };
 
   /// Changes the product's package id
-  virtual void ChangePackageId(int new_package_id) = 0;
+  virtual void ChangePackageId(int new_package_id) {};
 
  protected:
   const static int default_package_id_ = 1;

--- a/src/resource.h
+++ b/src/resource.h
@@ -95,7 +95,7 @@ class Resource {
   virtual int package_id() { return default_package_id_; };
 
   /// Changes the product's package id
-  virtual void ChangePackageId(int new_package_id) {};
+  virtual void ChangePackageId(int new_package_id = default_package_id_ ) {};
 
  protected:
   const static int default_package_id_ = 1;

--- a/src/resource.h
+++ b/src/resource.h
@@ -91,6 +91,12 @@ class Resource {
   /// @param res pointer to a resource to be absorbed by this resource
   virtual void Absorb(Ptr res) { throw Error("cannot absorb resource type " + this->type()); };
 
+  /// Returns the package id.
+  virtual int package_id() = 0;
+
+  /// Changes the product's package id
+  virtual void ChangePackageId(int new_package_id) = 0;
+
  protected:
   const static int default_package_id_ = 1;
  private:

